libpod/filters/containers.go

```diff
@@ -11,101 +10,133 @@ import (
-			}
-			return match
+			return util.StringMatchRegexSlice(c.Name(), filterValues)
 		}, nil
 	case "exited":
-		exitCode, err := strconv.ParseInt(filterValue, 10, 32)
-		if err != nil {
-			return nil, errors.Wrapf(err, "exited code out of range %q", filterValue)
+		var exitCodes []int32
+		for _, exitCode := range filterValues {
+			ec, err := strconv.ParseInt(exitCode, 10, 32)
+			if err != nil {
+				return nil, errors.Wrapf(err, "exited code out of range %q", ec)
+			}
+			exitCodes = append(exitCodes, int32(ec))
 		}
 		return func(c *libpod.Container) bool {
 			ec, exited, err := c.ExitCode()
-			if ec == int32(exitCode) && err == nil && exited {
-				return true
+			if err == nil && exited {
+				for _, exitCode := range exitCodes {
+					if ec == exitCode {
+						return true
+					}
+				}
 			}
 			return false
 		}, nil
 	case "status":
-		if !util.StringInSlice(filterValue, []string{"created", "running", "paused", "stopped", "exited", "unknown"}) {
-			return nil, errors.Errorf("%s is not a valid status", filterValue)
+		for _, filterValue := range filterValues {
+			if !util.StringInSlice(filterValue, []string{"created", "running", "paused", "stopped", "exited", "unknown"}) {
+				return nil, errors.Errorf("%s is not a valid status", filterValue)
+			}
 		}
 		return func(c *libpod.Container) bool {
 			status, err := c.State()
 			if err != nil {
 				return false
 			}
-			if filterValue == "stopped" {
-				filterValue = "exited"
-			}
 			state := status.String()
 			if status == define.ContainerStateConfigured {
 				state = "created"
 			} else if status == define.ContainerStateStopped {
 				state = "exited"
 			}
-			return state == filterValue
+			for _, filterValue := range filterValues {
+				if filterValue == "stopped" {
+					filterValue = "exited"
+				}
+				if state == filterValue {
+					return true
+				}
+			}
+			return false
 		}, nil
 	case "ancestor":
 		// This needs to refine to match docker
 		// - ancestor=(<image-name>[:tag]|<image-id>| ⟨image@digest⟩) - containers created from an image or a descendant.
 		return func(c *libpod.Container) bool {
-			containerConfig := c.Config()
-			if strings.Contains(containerConfig.RootfsImageID, filterValue) || strings.Contains(containerConfig.RootfsImageName, filterValue) {
-				return true
+			for _, filterValue := range filterValues {
+				containerConfig := c.Config()
+				if strings.Contains(containerConfig.RootfsImageID, filterValue) || strings.Contains(containerConfig.RootfsImageName, filterValue) {
+					return true
+				}
 			}
 			return false
 		}, nil
 	case "before":
-		ctr, err := r.LookupContainer(filterValue)
-		if err != nil {
-			return nil, errors.Errorf("unable to find container by name or id of %s", filterValue)
+		var createTime time.Time
+		for _, filterValue := range filterValues {
+			ctr, err := r.LookupContainer(filterValue)
+			if err != nil {
+				return nil, errors.Errorf("unable to find container by name or id of %s", filterValue)
```

Comment:
```suggestion
				return nil, errors.Wrapf(err, "unable to find container by name or id of %s", filterValue)
```

Reply:
Actually I can just return `err` here. The error from `LookupContainer` is `no container with name or ID %s found: no such container`. Wrapping this wouldn't provide more information